presto-parquet/src/main/java/io/prestosql/parquet/reader/LongColumnReader.java

```diff
@@ -29,7 +35,26 @@ public LongColumnReader(RichColumnDescriptor descriptor)
     protected void readValue(BlockBuilder blockBuilder, Type type)
     {
         if (definitionLevel == columnDescriptor.getMaxDefinitionLevel()) {
-            type.writeLong(blockBuilder, valuesReader.readLong());
+            // There is no easy way to support this for both iceberg and hive. hive does not have
+            // timestamp with timezone and iceberg stores timestamp with timezone as INT_64. Presto
+            // has its own way of storing timestamp with timezone and it has its own way to preserve
+            // the zone where as iceberg does not preserve zone info but stores just microseconds from 1970-01-01 00:00:00.000000 UTC
+            // in an 8-byte little-endian long. Given there are only 2 consumers right now we are hacking
+            // this piece and converting the long to UTC millisecond and writing it in presto timestamp with timezone's long format
+            // so presto-iceberg can handle timestamp with timezone type correctly
+            if (columnDescriptor.getPrimitiveType().getOriginalType() == OriginalType.TIMESTAMP_MICROS) {
+                final long utcMillis = TimeUnit.MICROSECONDS.toMillis(valuesReader.readLong());
+                if (type instanceof TimestampWithTimeZoneType) {
```

Comment:
Looks like it's not easily available at construction time, since this is created inside the Parquet reader. I'll extract out a `TimestampMicrosColumnReader` so we can at least skip the outer check, and add a TODO for the other one